src/gas/gas-util.test.ts

```diff
@@ -50,7 +54,75 @@ const mockEIP1559ApiResponses = [
   },
 ];
 
-describe('gas utils', () => {
+describe('gas-util', () => {
+  describe('normalizeGWEIDecimalNumbers', () => {
+    describe('given a number', () => {
+      it('should return a whole number as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(1)).toBe('1');
+        expect(normalizeGWEIDecimalNumbers(123)).toBe('123');
+        expect(normalizeGWEIDecimalNumbers(101)).toBe('101');
+        expect(normalizeGWEIDecimalNumbers(1234)).toBe('1234');
+        expect(normalizeGWEIDecimalNumbers(1000)).toBe('1000');
+      });
+
+      it('should return a decimal number greater than 1 as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(1.1)).toBe('1.1');
+        expect(normalizeGWEIDecimalNumbers(123.01)).toBe('123.01');
+        expect(normalizeGWEIDecimalNumbers(101.001)).toBe('101.001');
+        expect(normalizeGWEIDecimalNumbers(100.001)).toBe('100.001');
+        expect(normalizeGWEIDecimalNumbers(1234.567)).toBe('1234.567');
+      });
+
+      it('should return a decimal number less than 1 as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(0.1)).toBe('0.1');
+        expect(normalizeGWEIDecimalNumbers(0.01)).toBe('0.01');
+        expect(normalizeGWEIDecimalNumbers(0.001)).toBe('0.001');
+        expect(normalizeGWEIDecimalNumbers(0.567)).toBe('0.567');
+      });
+
+      it('should round a decimal number to 9 places before converting to a string', () => {
+        expect(normalizeGWEIDecimalNumbers(1.0000000162)).toBe('1.000000016');
+        expect(normalizeGWEIDecimalNumbers(1.0000000165)).toBe('1.000000017');
+        expect(normalizeGWEIDecimalNumbers(1.0000000199)).toBe('1.00000002');
+        expect(normalizeGWEIDecimalNumbers(1.9999999999)).toBe('2');
+        expect(normalizeGWEIDecimalNumbers(1.0000005998)).toBe('1.0000006');
+        expect(normalizeGWEIDecimalNumbers(123456.0000005998)).toBe(
+          '123456.0000006',
+        );
+        expect(normalizeGWEIDecimalNumbers(1.000000016025)).toBe('1.000000016');
+        expect(normalizeGWEIDecimalNumbers(1.0000000160000028)).toBe(
+          '1.000000016',
+        );
+        expect(normalizeGWEIDecimalNumbers(1.000000016522)).toBe('1.000000017');
+        expect(normalizeGWEIDecimalNumbers(1.000000016800022)).toBe(
+          '1.000000017',
+        );
+      });
+
+      it('should convert NaN to 0', () => {
+        expect(normalizeGWEIDecimalNumbers(NaN)).toBe('0');
+      });
+    });
+
+    describe('given a string', () => {
```

Comment:
I noticed that there were no rounding tests cases here. Seems that strings would need rounding as well.

I'm also wondering what this does when given a string that exceeds MAX_INT, or a number precise enough (i.e. with enough digits after the decimal point) that it can't be represented properly as a float in JavaScript.